python/desc/lsst_camera_readout/lsst_camera_readout.py

```diff
@@ -60,13 +69,26 @@ class ImageSource(object):
         Object containing the readout properties of the sensors in the
         focal plane, extracted from the segmentation.txt file.
     '''
-    def __init__(self, eimage_file, seg_file=None, sensor_id=None,
-                 add_read_noise=True):
+    def __init__(self, eimage_file=None, image_array=None, seg_file=None, 
```

Comment:
Thanks, Jim.  That is a good point about the problem with having a default for each of the parameters.  And your solution looks elegant.